src/go/k8s/pkg/resources/headless_service.go

```diff
@@ -118,3 +124,14 @@ func (r *HeadlessServiceResource) HeadlessServiceFQDN() string {
 		'.',
 		r.Key().Namespace)
 }
+
+func (r *HeadlessServiceResource) getAnnotation() map[string]string {
+	if !r.pandaCluster.Spec.ExternalConnectivity.Enabled {
```

Comment:
I think we should also not include this if subdomain is empty

Reply:
Agree, changed.